pkg/daemon/daemon.go

```diff
@@ -129,23 +131,25 @@ func New(
 	client snclientset.Interface,
 	kubeClient *kubernetes.Clientset,
 	mcClient *mcclientset.Clientset,
+	isHypershift bool,
```

Comment:
can we add a new PlatformType ? and avoid this extra Daemon attribute?

Reply:
@adrianchiris would you be ok with having this be a bool for now or does this comment rescind your lgtm?

Reply:
@adrianchiris i think no matter what you are going to have an extra variable in the daemon. It’s just a matter of if you want to have it be PlatformType which then means the if statements of !hypershift is more complex across code. I wanted to make that as clean as possible

let me know your thoughts on that! Thanks for your time!

Reply:
> @adrianchiris would you be ok with having this be a bool for now or does this comment rescind your lgtm?

seeing this PR has changed (quite a bit) since i last reviewed, ill need to review again so no lgtm on my side ATM.

@relyt0925 

if we can save the extra bool, id prefer that. In a few months maybe we will also have MegaShift (as an example ofc) would we then add another boolean ?

i see all usage of hypershift is in checks of this sort:
`utils.ClusterType == utils.ClusterTypeOpenshift && !dn.isHypershift`

so what i propose will simplify these checks since we will have just:
`utils.ClusterType == utils.ClusterTypeOpenshift `

for other places where you want to match on both a utility method can be added: `IsOpenshiftFamily()` or similar to handle it.

Thoughts ?

Reply:
I am not sure that necessarily it will always end up that way @adrianchiris : Openshift can have two "flavors": IPI/UPI and Hypershift. So I think it still makes sense. But will defer to what the community thinks is best.

Reply:
Ultimately there will still need to be multiple checks though (it cannot just be utils.ClusterType == utils.ClusterTypeOpenshift) 

I don't think we can save the extra var: I think we can replace the extra var (bool) for a more stateful var (which I sort of wanted to avoid as the extra state doesn't matter) but ultimately that var needs to be propogated for the proper openshift version detection.

Reply:
hi @relyt0925 what is ipi/upi ?

what i mean is that we have (generic example):

```
type ClusterType string

const ( 
  ClusterTypeOpenshift  ClusterType = "openshift"
  ClusterTypeHypershift ClusterType = "hypershift"
)

func IsOpenshiftFamilty(ct ClusterType) bool {
  return ct == ClusterTypeOpenshift || ct == ClusterTypeHypershift
}

// do things only on openshift
if ct == ClusterTypeOpenshift {
  // do things ...
}

// do things only on hyperhift
if ct == ClusterTypeHypershift {
  // do things ...
}

``


or we should have just Openshift as cluster type and then have a different attribute for openshift flavor.

another thing which would be beneficial in regards to code tidiness is maybe to put all openshift related attr, 
like openshiftFlavor/ishypershift , mcpName, mcClient into its own struct (OpenshiftContext ?) and have this as an attr of the daemon.

lets hear other  opinions on the matter.

Reply:
@adrianchiris IPI or UPI is way to install openshift. I don't think that's really relevant for our discussion.

What @relyt0925 is saying is that within openshift, there are different "sub types".

I do like the idea of refactoring everything openshift related into an OpenshiftContext struct, but in that case we'd still need a top level "isOpenshift" (or clusterType that can distinguish between vanilla k8s, openshift, insert_your_own_new_falvor, ...) and with all the related openshift stuff inside the OpenshiftContext struct which should only be looked at if the top level clusterType is openshift. I guess that's what you mean @adrianchiris ?

Reply:
> IPI or UPI is way to install openshift. 

ack thanks for clarifying !

> I guess that's what you mean
yep

Reply:
@relyt0925 Can you refactor into OpenshfitContext as a first commit, and followed by adding the isHypershift inside there in an additional commit to keep the history clean?

Reply:
sure.

Reply:
refactored based on comments. 1st commit contains new struct holding necessary openshift metadata.